### PR TITLE
✨ Feat(#6902) UI - Add condition support for policy rules

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
### Describe your changes :

I worked on #6902 and added the effect field in the rules card.

#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/59080942/186400356-fb0a55a6-74c5-4ff3-9f6e-919e8c2158f9.mov

cc : @sureshms @harshach 

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/pages/PoliciesPage/RuleForm/RuleForm.tsx [hunk 1]
```diff
@@ -37,6 +43,14 @@ const RuleForm: FC<RuleFormProps> = ({ ruleData, setRuleData }) => {
     []
   );
 
+  const [policyFunctions, setPolicyFunctions] = useState<Function[]>([]);
+
+  const [conditionOptions, setConditionOptions] = useState<
+    { label: string; value: string }[]
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/PoliciesPage/RuleForm/RuleForm.tsx:49]: Don't we have generatedTypes?

File: openmetadata-ui/src/main/resources/ui/src/pages/PoliciesPage/RuleForm/RuleForm.tsx [hunk 2]
```diff
@@ -91,6 +105,30 @@ const RuleForm: FC<RuleFormProps> = ({ ruleData, setRuleData }) => {
     return option;
   }, [ruleData.resources, policyResources]);
 
+  const getConditionOptions = (funtions: Function[]) => {
+    return funtions.reduce(
+      (prev: { label: string; value: string }[], curr: Function) => {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/PoliciesPage/RuleForm/RuleForm.tsx:110]: If not generated then create one like:

type OptionT = {
...
}

File: openmetadata-ui/src/main/resources/ui/src/pages/PoliciesPage/RuleForm/RuleForm.tsx [hunk 3]
```diff
@@ -100,10 +138,43 @@ const RuleForm: FC<RuleFormProps> = ({ ruleData, setRuleData }) => {
...
+    const defaultErrorText = 'Condition is invalid';
+    if (ruleData.condition) {
+      try {
+        const response = await validateRuleCondition(ruleData.condition);
+
+        /**
+         * If request is successful then we will only get response as success without any data
+         * So, here we have to check the response status code
+         */
+        [200, 204].includes(response.status)
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/PoliciesPage/RuleForm/RuleForm.tsx:160]: We might have this code from AxiosResponseCode can you chech?

File: openmetadata-ui/src/main/resources/ui/src/pages/PoliciesPage/RuleForm/RuleForm.tsx (hunk 1 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/pages/PoliciesPage/RuleForm/RuleForm.tsx:49]: No, This is for antd Autocomplete option component.

File: openmetadata-ui/src/main/resources/ui/src/pages/PoliciesPage/RuleForm/RuleForm.tsx (hunk 3 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/pages/PoliciesPage/RuleForm/RuleForm.tsx:160]: Sure.

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**